### PR TITLE
Use "git rev-parse --absolute-git-dir" in tools scripts.

### PR DESCRIPTION
**Pull request by ywkaras**
The alternate way of getting this, that we are currently using, doesn't work in cmake version 3.28.

### CODE REVIEW

**Review by JosiahWI — CHANGES_REQUESTED** [review]
Seems like a good improvement, but could we change the variable name?

File: tools/autopep8.sh
Comment (JosiahWI) [inline tools/autopep8.sh:42]: The variable name is no longer correct, as this will correspond to `$GIT_DIR` and **not** to `$GIT_COMMON_DIR`. I think this should be fine, but could confirm that and update the name to reflect this?


> --absolute-git-dir
Like --git-dir, but its output is always the canonicalized absolute path.

**Review by JosiahWI — CHANGES_REQUESTED** [review]

File: tools/autopep8.sh
Comment (ywkaras) [reply tools/autopep8.sh:42]: So is it a cosmetic issue, that we follow some convention for naming this issue?  Or, do you mean we want to make it overridable from the process environment, like we do for AUTOPEP8_VENV on the next line?

Comment (JosiahWI) [reply tools/autopep8.sh:42]: It's a readability issue. It's like the difference between `PROJECT_SOURCE_DIR` and `CMAKE_SOURCE_DIR` in CMake. They're not the same thing and in edge cases the difference matters. Your commit introduces a behavioral change to the script, and the name of the `GIT_COMMON_DIR` variable is lying about what itself. It's not the git common dir anymore.

Comment (JosiahWI) [reply tools/autopep8.sh:42]: > GIT_COMMON_DIR
If this variable is set to a path, non-worktree files that are normally in $GIT_DIR will be taken from this path instead. Worktree-specific files such as HEAD or index are taken from $GIT_DIR. See [gitrepository-layout[5]](https://git-scm.com/docs/gitrepository-layout) and [git-worktree[1]](https://git-scm.com/docs/git-worktree) for details. This variable has lower precedence than other path variables such as GIT_INDEX_FILE, GIT_OBJECT_DIRECTORY…​

Comment (ywkaras) [reply tools/autopep8.sh:42]: But, given that it's not being put into the environment, is GIT_DIR also not an appropriate name?

Comment (JosiahWI) [reply tools/autopep8.sh:42]: I would be fine with `GIT_DIR`. Would shadowing the environment variable be harmful? The previous name was shadowing one, so it would be consistent with the current pattern to simply rename to `GIT_DIR` I think.

**Review by JosiahWI — COMMENTED** [review]
Thank you for changing the variable name. Looks good to me and makes the scripts easier to follow. Will let a non-Yahoo person approve.

**Review by JosiahWI — COMMENTED** [review]
Dismiss pending change request as it has been addressed.

**Review by JosiahWI — APPROVED** [review]
I think GitHub has a small issue. Giving explicit approval to dismiss the change request. PR still needs approval from a non-Yahoo person.

**Review by cmcfarlen — APPROVED** [review]
This works fine on macOS

**Comment by cmcfarlen** [comment]
Cherry-picked to v10.0.x

**Pull request merged**
